### PR TITLE
The State Machine implemented

### PR DESCRIPTION
**Pull request by AlexanderSuprunenko**

### CODE REVIEW

File: include/state_machine.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        }
+    }
+
+private:
+    void init_table();
+    State state() const { return m_state; }
+    State state(State state) { return m_state = state; }
+    St status(BaseTaskPtr bt) const { return bt->getLastStatus(); }
+
+    void process(BaseTaskPtr bt)
```
Comment (bitkis) [inline include/state_machine.h]: Don't we need some logging here??

Comment (AlexanderSuprunenko) [reply include/state_machine.h]: Not sure. On one hand the messages would accompany others logging messages. On the other, they would be useful if we forgot to log such others, and they should be of special categorie, which we have no yet, to not overwhelm the log.

Comment (bitkis) [reply include/state_machine.h]: please make this PR vs. development/v0.0.2

**Comment by bitkis** [comment]
please make this PR vs. development/v0.0.2

File: src/task.cpp [hunk 2]
```diff
@@ -10,6 +11,131 @@ namespace graft {
...
 
+const StateMachine::Guard StateMachine::has(Router::Handler H3::* act)
+{
+    return [act](BaseTaskPtr bt)->bool
+    {
+        return (bt->getHandler3().*act != nullptr);
+    };
+}
+
+const StateMachine::Guard StateMachine::hasnt(Router::Handler H3::* act)
```
[the remark this answers is not in the document]
Comment (AlexanderSuprunenko) [reply src/task.cpp]: No. A lambda is returned. ! for lambda has different meaning.

**Pull request merged**